DataRepo/utils/exceptions.py

```diff
@@ -193,18 +197,23 @@ def __init__(
         consistent_field,
         existing_value,
         differing_value,
+        rownum=None,
         message=None,
     ):
         if not message:
+            rowmsg = ""
+            if rownum:
+                rowmsg = f"on row {rownum} "
             message = (
-                f"Conflicting values encountered in {type(rec).__name__} record [{str(rec)}] for the "
+                f"Conflicting values encountered {rowmsg}in {type(rec).__name__} record [{str(rec)}] for the "
```

Comment:
Does `rowmsg` always contain a trailing space?
```suggestion
                f"Conflicting values encountered {rowmsg} in {type(rec).__name__} record [{str(rec)}] for the "
```

Reply:
The spacing is correct.  Either it contains text with a tailing space or it is an empty string (i.e. it should not have a trailing space).